DOCS-#3792, DOCS-#2410: Improve "Getting Started" section (#3965)
modin-project/modin: docs/getting_started/troubleshooting.rst

This is a list of the most frequently encountered issues when using Modin. Some of these
are working as intended, while others are known bugs that are being actively worked on.

> Warning during execution: ``defaulting to pandas``
> """""""""""""""""""""""""""""""""""""""""""""""""""
> Please note, that while Modin covers a large portion of the pandas API, not all functionality is implemented. For methods that are not yet implemented, such as ``asfreq``, you may see the following:

Reviewer: ```suggestion
Warning during execution: ``defaulting to pandas``
""""""""""""""""""""""""""""""""""""""""""""""""""

Please note, that while Modin covers a large portion of the pandas API, not all functionality is implemented. For methods that are not yet implemented, such as ``asfreq``, you may see the following:
```

  pd_df = pd_df.set_index(index_col_name)
  pd_df.index.name = None

Using Modin with python multiprocessing
"""""""""""""""""""""""""""""""""""""""

> Note that Modin can not be used in conjunction with python multiprocessing. For example, if the Ray engine is selected, it can not be used alongside python multiprocessing.

Reviewer: ```suggestion
We strongly recommend not to mix the use of Modin with Ray or Dask engine selected
in conjunction with python multiprocessing because that can lead to undefined behavior.
One of such examples is shown below:

.. code-block:: python
  import modin.pandas as pd
  
  # Ray engine is used by default
  df = pandas.DataFrame([1, 2, 3])
  
  def f(arg):
    return df + arg

  if __name__ == '__main__':
    from multiprocessing import Pool
    
    with Pool(5) as p:
        print(p.map(f, [1]))

Even if this example may work on your machine, we do not recommend similar scenarios.
The python multiprocessing will cause conflicts with excessive resource use
by launching duplicated Ray clusters on the same machine.
```

Author: This looks good, thanks @YarShev !

  pd_df = pd_df.set_index(index_col_name)
  pd_df.index.name = None

Using Modin with python multiprocessing
"""""""""""""""""""""""""""""""""""""""

We strongly recommend not to mix the use of Modin with Ray or Dask engine selected
in conjunction with python multiprocessing because that can lead to undefined behavior.
One of such examples is shown below:

.. code-block:: python
  import modin.pandas as pd
  
  # Ray engine is used by default
  df = pandas.DataFrame([1, 2, 3])
  
  def f(arg):
    return df + arg

  if __name__ == '__main__':
    from multiprocessing import Pool
    
    with Pool(5) as p:
        print(p.map(f, [1]))

Even if this example may work on your machine, we do not recommend similar scenarios.
The python multiprocessing will cause conflicts with excessive resource use
by launching duplicated Ray clusters on the same machine.

> Common errors
> ---------------

Reviewer: ```suggestion
Common errors
-------------
```

  pd_df = pd_df.set_index(index_col_name)
  pd_df.index.name = None

Common errors
---------------

> Error during execution: ``ArrowIOError: Broken Pipe``
> """""""""""""""""""""""""""""""""""""""""""""""""""""
> 
> One of the more frequently encountered issues is an ``ArrowIOError: Broken Pipe``. This
> error can happen in a couple of different ways. One of the most common ways this is
> encountered is from pressing **CTRL + C** sending a ``KeyboardInterrupt`` to Modin. In
> Ray, when a ``KeyboardInterrupt`` is sent, Ray will shutdown. This causes the
> ``ArrowIOError: Broken Pipe`` because there is no longer an available plasma store for
> working on remote tasks. This is working as intended, as it is not yet possible in Ray
> to kill a task that has already started computation.
> 
> The other common way this ``Error`` is encountered is to let your computer go to sleep.
> As an optimization, Ray will shutdown whenever the computer goes to sleep. This will
> result in the same issue as above, because there is no longer a running instance of the
> plasma store.
> 
> **Solution**
> 
> Restart your interpreter or notebook kernel.
> 
> **Avoiding this Error**
> 
> Avoid using ``KeyboardInterrupt`` and keeping your notebook or terminal running while
> your machine is asleep. If you do ``KeyboardInterrupt``, you must restart the kernel or
> interpreter.
> 
> Error during execution: ``ArrowInvalid: Maximum size exceeded (2GB)``
> """""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""""
> 
> Encountering this issue means that the limits of the Arrow plasma store have been
> exceeded by the partitions of your data. This can be encountered during shuffling data
> or operations that require multiple datasets. This will only affect extremely large
> DataFrames, and can potentially be worked around by setting the number of partitions.
> This error is being actively worked on and should be resolved in a future release.
> 
> **Solution**
> 
> .. code-block:: python
> 
>   import modin.pandas as pd
>   pd.DEFAULT_NPARTITIONS = 2 * pd.DEFAULT_NPARTITIONS
> 
> This will set the number of partitions to a higher count, and reduce the size in each.
> If this does not work for you, please open an issue_.
> 

Reviewer 2: ```suggestion
```

These two issues don't happen anymore, they have been fixed.
